src/resource/resource.c

```diff
@@ -451,8 +451,28 @@ int sid_resource_create_io_event_source(sid_resource_t *res, sid_resource_event_
 
 static int _sd_signal_event_handler(sd_event_source *sd_es, const struct signalfd_siginfo *si, void *data)
 {
+	switch (si->ssi_signo) {
+		case SIGTERM :
+			sd_event_exit(((sid_resource_t *) sd_es)->sd_event_loop, 0);
+			break;
+		case SIGINT:
+			break;
+		case SIGPIPE:
+			break;
+		case SIGHUP: /* TODO: Reload config on SIGHUP? */
+			break;
+		case SIGCHLD:
+			break;
+		default:
+			break;
+	};
```

Comment:
Well, I think we should move this out of `resource.c` to the caller. The `resource.c` is generic reusable code, the caller itself should register signals and possible handlers it needs. What if someone wants to define a resource and use resource API and doesn't want the SIGTERM to exit the event loop, but do something else. So for the `SIGTERM` (and other signals), we need to register a handler directly in the caller instead (which is `src/resource/sid.c` here).

Then if the caller wants to exit the event loop, it can use the `sid_resource_exit_event_loop` resource API function for that.

Reply:
Ok - will do

Reply:
moved to resource/sid.c - I think that is what you were asking